src/Stratis.Bitcoin/Configuration/NodeSettings.cs

```diff
@@ -261,25 +262,24 @@ public static NodeSettings Default(Network network, ProtocolVersion protocolVers
         /// Creates the configuration file if it does not exist.
         /// </summary>
         /// <param name="features">The features for which to include settings in the configuration file.</param>
-        public void CreateDefaultConfigurationFile(List<IFeatureRegistration> features)
+        public void CreateDefaultConfigurationFile(List<Type> features, IServiceProvider serviceProvider)
```

Comment:
Why do we need `serviceProvider` in the method signature?

Reply:
Its a way for the static settings class methods to gain access to already created settings object instances of the relevant full node.